rxandroid/src/main/java/rx/android/lifecycle/LifecycleHelper.java

```diff
@@ -0,0 +1,92 @@
+package rx.android.lifecycle;
+
+import android.app.Activity;
+import android.app.Application;
+import android.os.Bundle;
+
+import rx.subjects.BehaviorSubject;
+
+/**
+ * This instance lives within the scope of an {@code Activity} in order to forward Lifecycle events
+ * to {@link rx.subjects.BehaviorSubject}
+ *
+ * Thread safe, since it's <i>Thread-confined</i> to the main Thread.
+ *
+ * @author Nabil Hachicha.
+ */
+class LifecycleHelper implements Application.ActivityLifecycleCallbacks {
+    private final Activity activityToMonitor;
```

Comment:
I've always thought it would be cool to have the `Subject` output all events, then use `Observable.filter()` to limit which ones you care about. Would get rid of all the `==` checks throughout the rest of this class.